block_manipulation_service/src/main/java/com/stnetix/ariaddna/blockmanipulation/BlockGenerate.java

```diff
@@ -0,0 +1,107 @@
+package com.stnetix.ariaddna.blockmanipulation;
+
+import com.stnetix.ariaddna.commonutils.datetime.DateTime;
+import com.stnetix.ariaddna.commonutils.xmlparser.XmlParser;
+import com.stnetix.ariaddna.commonutils.xmlparser.exception.XmlParserException;
+import com.stnetix.ariaddna.commonutils.xmlparser.handlers.BlockManipulationSettingHandler;
+import com.stnetix.ariaddna.localstoragemanager.LocalService.LocalService;
+import com.stnetix.ariaddna.vufs.BusinessObjects.Block;
+import com.stnetix.ariaddna.vufs.BusinessObjects.Metafile;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.*;
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.util.Date;
+import java.util.HashMap;
+
+/**
+ * Created by LugovoyAV on 15.02.2018.
+ */
+@Service
+public class BlockGenerate {
+    private final int defaultBlockSize = 5242880;
+    private int blockSize;
+
+    private HashMap<File,Long> files = new HashMap<>();
+
+    public int getBlockSize() {
+        return blockSize;
+    }
+
+    public BlockGenerate() {
+        if (!loadSettings())
+            blockSize = defaultBlockSize;
+    }
+
+    @Autowired
+    private LocalService localService;
+
+    public boolean loadSettings()
+    {
+        BlockManipulationSettingHandler handler = null;
+        File file = null;
+        try {
+
+            String resource = "settings.xml";
+            URL settingURL = getClass().getClassLoader().getResource(resource);
+            file = new File(settingURL.getFile());
+            handler = (BlockManipulationSettingHandler) new XmlParser(file, new BlockManipulationSettingHandler()).getHandler();
+            blockSize = handler.getBlockSize();
+            return true;
+        } catch (XmlParserException e) {
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+    public Block getNextBlock(Metafile metaFile) throws IOException {
+        return createNextBlock(metaFile.getVersion(),metaFile.getFileUuid());
+    }
+
+    private Block createNextBlock(String version, String fileUuid) throws IOException {
+        long offset = -1;
+        File localFile = localService.getLocalFileByUuid(fileUuid);
+        if (!localFile.isFile())
+            throw new IllegalArgumentException("Can't get block for directory");
+
+
+        //TODO we should be call restoreFileContext(metafile). We add this method in the features tasks.
```

Comment:
If you want store some issue\feature for team please create issue on github. ;)

Reply:
create issue https://github.com/StnetixDevTeam/ariADDna/issues/131